lib/msal-react/docs/migration-guide.md

```diff
@@ -0,0 +1,243 @@
+
+```javascript
+import { PublicClientApplication } from "@azure/msal-browser";
+import { MsalProvider } from "@azure/msal-react";
+
+const pca = new PublicClientApplication(config);
+
+function App() {
+    return (
+        <MsalProvider instance={pca}>
+            <YourAppComponents />
+        </MsalProvider>
+    );
+}
+```
+
+General notes about the `MsalProvider` component:
+
+- All components that need access to authentication state or hooks/components exposed by `msal-react` must have an `MsalProvider` higher up in the component tree, therefore it is recommended that `MsalProvider` be rendered as close to the root as possible.
+- Your app should not render more than 1 `MsalProvider` component on any given page.
+- We do not recommend initializing `PublicClientApplication` inside a component due to the possiblity of re-renders
+
+### Protecting your components
+
+In `react-aad-msal` components are protected using the `AzureAD` component or `withAuthentication` HOC which wraps your component with `AzureAD` under the hood. The `AzureAD` component will only render child components if a user is authenticated and optionally initiate a login if no user is authenticated. The options used for login (e.g. Scopes, whether to use popup or redirect, etc.) are specified earlier, when creating the `authProvider` prop.
+
+```javascript
+import { MsalAuthProvider } from "react-aad-msal";
+
+const authProvider = new MsalAuthProvider(config, authenticationParameters, options);
+
+function App() {
+    return (
+        <AzureAD provider={authProvider} forceLogin={true}>
+            <span>Only authenticated users can see me.</span>
+        </AzureAD>
+    );
+}
+```
+
+`msal-react`, on the other hand, gives developers more control over what they want to display to whom.
+
+- The `AuthenticatedTemplate` component will render children if a user is authenticated
+- The `UnauthenticatedTemplate` component will render children if a user is not authenticated
+- The `MsalAuthenticationTemplate` component will automatically initiate a login if a user is unauthenticated and then render children once a user is authenticated.
+
+```javascript
+import { PublicClientApplication } from "@azure/msal-browser";
+import { MsalProvider, AuthenticatedTemplate, UnauthenticatedTemplate, MsalAuthenticationTemplate } from "@azure/msal-react";
+
+const pca = new PublicClientApplication(config);
+
+function App() {
+    return (
+        <MsalProvider instance={pca}>
+            <AuthenticatedTemplate>
+                <span>Only authenticated users can see me.</span>
+            </AuthenticatedTemplate>
+            <UnauthenticatedTemplate>
+                <span>Only unauthenticated users can see me.</span>
+            </UnauthenticatedTemplate>
+            <MsalAuthenticationTemplate interactionType="popup" authenticationRequest={request}>
```

Comment:
Can this use the enum for interaction type, and show an example `request` object?

Reply:
Done